### PR TITLE
iOS SDK notification handling package

### PR DESCRIPTION
**Pull request by dangeross**
Add to the package definitions the added notification handling source files and dependencies.

Breez SDK PR https://github.com/breez/breez-sdk/pull/822
C-Breez PR https://github.com/breez/c-breez/pull/803

### CODE REVIEW

**Review by erdemyerebasmaz — COMMENTED** [review]
LGTM

**Comment by dangeross** [comment]
@cnixbtc it would be great to get your input if you see any issues with these podspec/package changes

File: Package.swift [hunk 1]
```diff
@@ -7,15 +7,17 @@ let package = Package(
     name: "bindings-swift",
     platforms: [
         .macOS(.v12),
-        .iOS(.v11),
+        .iOS(.v14),
     ],
     products: [
-        .library(name: "BreezSDK", targets: ["breez_sdkFFI", "BreezSDK"]),
+        .library(name: "BreezSDK", targets: ["breez_sdkFFI", "BreezSDK", "BreezSDKNotification"]),
```
Comment (cnixbtc) [inline Package.swift:13]: Is the `BreezSDKNotification` target generated automatically or do we need to declare it explicityl below in the targets array? 🤔

Comment (dangeross) [reply Package.swift:13]: Good point, I think I'm thinking about it backwards. I should have 3 targets and 1 library:
```
targets: [
        .binaryTarget(name: "breez_sdkFFI", url: "https://github.com/breez/breez-sdk-swift/releases/download/0.3.1/breez_sdkFFI.xcframework.zip", checksum: "fcda9a87b512fc0ab81619096d24db6e54b622c18cded6d1d78db95f05c10265"),
        .target(name: "BreezSDK", dependencies: ["breez_sdkFFI"]),
        .target(name: "BreezSDKNotification", dependencies: ["BreezSDK", "KeychainAccess", "XCGLogger"]),
    ]
```

**Comment by erdemyerebasmaz** [comment]
Does this require a change on 
- https://github.com/breez/breez-sdk/pull/834

?

File: Package.swift (hunk 1 above)
Comment (erdemyerebasmaz) [reply Package.swift:13]: Does this require a change on 
- https://github.com/breez/breez-sdk/pull/834

?

Comment (dangeross) [reply Package.swift:13]: No, it should pickup the target source files by directory name "Sources/BreezSDKNotification"

**Review by erdemyerebasmaz — CHANGES_REQUESTED** [review]
Latest changes look good 👏 

There's one small todo item before merging.

As discussed on Slack, we had upgraded to iOS 14 due to `XCGLogger` dependency and since we no longer depend on it please verify if downgrading back to iOS 11 is viable.

**Review by erdemyerebasmaz — APPROVED** [review]

**Pull request merged**